app/src/main/java/com/example/tmdb/data/FavoritesViewModel.kt

```diff
@@ -0,0 +1,13 @@
+package com.example.tmdb.data
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.SharedFlow
+import org.koin.core.component.KoinComponent
+
+
+class FavoritesViewModel(var movieRespository: MovieRepository): ViewModel(), KoinComponent {
+    fun getFavoriteMovies(): SharedFlow<MutableList<MovieItemViewState>>{
+        return movieRespository.loadFavoriteMovies()
+    }
```

Comment:
You don't need extend class with `KoinComponent`, because you using constructor injection already.